### ISSUE TITLE
Font-locking lost after following Howm link

### ISSUE THREAD

**Issue by mmarshall540** [kaorahi/howm#22]
Hello, I'm noticing that some of the font-locking in non-Howm buffers gets disabled after following a Howm go-to link.  (I might be using the wrong terminology here.  What I mean is that the colors and styling of characters is not happening as one would expect.  Whether "font-locking" is the actual mechanism used in this instance, I'm not sure.)

I first noticed this in "\*Completions\*" buffers, but it also happens in some other buffers, such as the one created by `list-faces-display`.

It's a very stark change when using the Marginalia package, which adds a lot of information to Completions.  But the issue is reproducible on my end even without Marginalia.

I'm running Emacs 29.4 on debian Bookworm.

Here's a procedure for reproducing the described behavior.

1. First do `mkdir -p ~/emacs-howm/howm-test`. "~/emacs-howm" will be a temporary Emacs configuration, and the "howm-test" directory inside of it will be the `howm-directory`.
2.  Put the following in "~/emacs-howm/init.el".
```emacs-lisp
(setopt custom-enabled-themes '(modus-vivendi))

(require 'package)
(add-to-list 'package-archives '("melpa" . "https://melpa.org/packages/") t)
(package-initialize)

(use-package howm
  :ensure t
  :init
  (setq howm-directory "~/emacs-howm/howm-test")
  (setq howm-home-directory howm-directory))
```
(The modus-vivendi theme makes it easier to see what's happening, but the effect is the same without a theme.)
3. Open emacs from a command line by doing `emacs --init-directory=~/emacs-howm`.
4. Create two notes, with the second one linking to the first.
```text
= Test Note 1
[2024-12-15 17:06] 

This is a test.
```
And ...
```text
= Test Note 2
[2024-12-15 17:06] >>> ~/emacs-howm/howm-test/2024/12/2024-12-15-170602.txt

>>> Test Note 1
```

5. Close Emacs and start it again from the command line, as in step 3.
6. Press "C-c , a" to open the list of all notes.
7. Move the cursor to the line for "Test Note 2" and press RET to open Test Note 2.
8. At this point, you can confirm that the font-locking in the "\*Completions\*" buffer still works by pressing "C-x b TAB".  You will notice that the first character of each of the possible completions is a shade of blue, and the line above them stating "6 possible completions:" is a dark grey.
9. Press "C-g" to cancel the switch-to-buffer command.
10. Back in Test Note 2, move the cursor to the link text, ">>> Test Note 1" and press RET.  We are now back in the Howm Summary buffer.
11. Repeat the check from step 8.  Now the completions and the line above them are all the same shade of white.  Only the key notation for "RET" in the second line of the help text is colored now.
12. Also, if you do "M-x list-faces-display RET", you'll notice that none of the listed faces are displayed in their face.

Thank you for your attention. And thank you for this wonderful piece of software.

**Comment by kaorahi** [comment]
Thank you for the detailed report. It's very helpful.

This issue is likely caused by 280eead in https://github.com/kaorahi/howm/issues/11 . To confirm, could you test if the code below fixes the issue?

```
(setq riffle-protected-localvar-prefixes
  '("action-lock-" "howm-" "illusion-" "riffle-"))
```

**Comment by mmarshall540** [comment]
Yes, that seems to prevent it from happening as long as I evaluate that code beforehand.

**Comment by kaorahi** [comment]
Glad it worked. I'd like to test the fix locally for a week or so before officially applying it.

**Comment by kaorahi** [comment]
fixed in 9f0145b
